### DISCUSSION TITLE
Error when sending and resending user and org invitations

### DISCUSSION THREAD

**Discussion by strausmann** [dani-garcia/vaultwarden#5068] [Q&A] [answered]
### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.32.1
* Web-vault version: v2024.6.2c
* OS/Arch: linux/x86_64
* Running within a container: true (Base: Alpine)
* Environment settings overridden: true
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: MySQL
* Database version: 10.11.6-MariaDB
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** DOMAIN, HIBP_API_KEY, SIGNUPS_ALLOWED, SIGNUPS_VERIFY, SIGNUPS_VERIFY_RESEND_TIME, SIGNUPS_VERIFY_RESEND_LIMIT, SIGNUPS_DOMAINS_WHITELIST, EMERGENCY_ACCESS_ALLOWED, ADMIN_TOKEN, INVITATION_ORG_NAME, IP_HEADER, REQUIRE_DEVICE_EMAIL, YUBICO_CLIENT_ID, YUBICO_SECRET_KEY, _ENABLE_DUO, DUO_IKEY, DUO_SKEY, DUO_HOST, SMTP_HOST, SMTP_SECURITY, SMTP_PORT, SMTP_FROM, SMTP_FROM_NAME, SMTP_USERNAME, SMTP_PASSWORD, HELO_NAME

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": true,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_max_note_size": 10000,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "auth_request_purge_schedule": "30 * * * * *",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 20,
  "database_timeout": 30,
  "database_url": "*****://*********************************************************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://********************",
  "domain_origin": "*****://********************",
  "domain_path": "",
  "domain_set": true,
  "duo_context_purge_schedule": "30 * * * * *",
  "duo_host": "api-c605ae7c.duosecurity.com",
  "duo_ikey": "DI838TFXWLJOZFLT6CAA",
  "duo_skey": "***",
  "duo_use_iframe": false,
  "email_2fa_auto_fallback": false,
  "email_2fa_enforce_on_verified_invite": false,
  "email_attempts_limit": 3,
  "email_change_allowed": true,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "enable_websocket": true,
  "enforce_single_org_with_reset_pw_policy": false,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "experimental_client_feature_flags": "fido2-vault-credentials",
  "extended_logging": true,
  "helo_name": "**********************",
  "hibp_api_key": "***",
  "http_request_block_non_global_ips": true,
  "http_request_block_regex": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "increase_note_size_limit": false,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Strausmann Vault",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": "/logs/vaultwarden.log",
  "log_level": "debug",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": true,
  "org_groups_enabled": true,
  "password_hints_allowed": true,
  "password_iterations": 600000,
  "push_enabled": true,
  "push_identity_uri": "https://identity.bitwarden.com",
  "push_installation_id": "***",
  "push_installation_key": "***",
  "push_relay_uri": "https://push.bitwarden.com",
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": true,
  "signups_domains_whitelist": "**************,**************,**********************,*****************,************,************,*************",
  "signups_verify": true,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "****************************",
  "smtp_from_name": "Strausmann Vault",
  "smtp_host": "*******************",
  "smtp_password": "***",
  "smtp_port": 465,
  "smtp_security": "force_tls",
  "smtp_ssl": true,
  "smtp_timeout": 15,
  "smtp_username": "****************************",
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": null,
  "user_send_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "yubico_client_id": "59757",
  "yubico_secret_key": "***",
  "yubico_server": null
}
```
</details>

### Vaultwarden Build Version

v1.32.1

### Deployment method

Official Container Image

### Custom deployment method

vaultwarden/server:alpine

### Reverse Proxy

traefik 3.1.5

### Host/Server Operating System

Linux

### Operating System Version

Ubuntu 22.04

### Clients

Web Vault

### Client Version

Edge Version 129.0.2792.79 (Offizielles Build) (64-Bit)

## Resend User Invite
1. Login in to the Vaultwarden Admin
2. Switch to Users and click to the link "Resend Invite"
3. I got then the Error Message "Error resend invite, handlebars"

## Resend Org Invite
1. Login in to the Web Vault
2. Switch to Admin Console -> Members and Resend a Org Invite
3. I got then the Error Message "Ein Fehler ist aufgetreten. Handlebars"

### Expected Result

That a new invitation is sent and no error message is displayed.

### Actual Result

Error message is displayed, but no entry in the log, despite activated debug mode and log output on debug.

### Logs

```text
Nothing to see in the Logs files for this issue
```

### Screenshots or Videos

Resend User Invite
![image](https://github.com/user-attachments/assets/66b4a41f-4499-4097-938a-9104e5568d5b)

Resend Org Invite
![image](https://github.com/user-attachments/assets/df6d090b-6da5-42fd-923c-6e9dae5192fd)

### Additional Context

I had used the German email templates as a test. However, this has already been reversed.

Emails are sent if, for example, you log in with a username/password but have not completed a 2FA. The test emails are also successfully sent by the Vaultwarden admin to a test email address.

Mails are sent via Mailgun as SMTP Relay.

**Comment by BlackDex** [comment]
You are probably using translated or customized template files.
As mentioned in the release notes there were several of these templates changed.
Please update those templates to match with the new versions.

https://github.com/dani-garcia/vaultwarden/tree/1.32.1/src/static/templates/email

**Comment by strausmann** [comment]
@BlackDex As you can see from my description, I no longer use the templates.

**Comment by BlackDex** [comment]
I doubt that, because then you wouldn't have created this report.
How did you made sure it isn't using it anymore?
If you have a folder called `templates` in your `data` folder, then it is still using that.

Rename the folder, restart Vaultwarden and check again.

**Comment by strausmann** [comment]
This is the content of my “data” folder, there is no templates folder.

![image](https://github.com/user-attachments/assets/06f927de-16f1-48dc-926e-8f0716b6e945)

I have restarted Vaultwarden again, there is the same problem.

![image](https://github.com/user-attachments/assets/1729eae0-67dd-4ad2-9dcd-7e968c5d0386)

This is the output of the /data order in the container itself. I have also restarted the stack so that the container is also recreated.

I have also copied the complete Vaultwarden original “[templates](https://github.com/dani-garcia/vaultwarden/tree/1.32.1/src/static/templates)” folder into my “data” folder.

I still have the same problem here too.

**Comment by BlackDex** [comment]
I just tested it, and it works just fine.

Are you using a container build by us, or did you created the container your self?
If that is the case, and you built-in the changed templates, it could still have an issue, though that should have been fixed if you used the templates folder from the repo.

I have really no clue why it doesn't work for you.
Also, it should report something in the logs regarding this.
Maybe try to set the `LOG_LEVEL=debug` to see what is happening.

**Reply by strausmann** [reply, marked as answer]
Now it works again as desired without any problems. I have completely deleted the entire folder structure on the Ubuntu server once and restored it from the backup. Then restarted the Docker stack again.
The log level was set to DEBUG and I used the image vaultwarden/server:alpine. The current image was also used here via DOCKER PULL.

Thank you very much for your effort.

**Reply by BlackDex** [reply]
Very strange. Almost looks like it wasn't restarted, and some residual inode's were still able to be seen and fetched and there for the template still loaded or something?

Glad it is resolved :)
